lib/regexpcache/regexpcache.go

```diff
@@ -0,0 +1,199 @@
+package regexpcache
+
+import (
+	"container/heap"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/fasttime"
+)
+
+// Cache caches any values with limited number of entries
+// items evicted in LRU order
+type Cache struct {
+	requests          uint64
+	misses            uint64
+	cleanerMustStopCh chan struct{}
+	cleanerStoppedCh  chan struct{}
+	mu                sync.Mutex
+	m                 map[string]*cacheEntry
+	perKeyMisses      map[string]int
+	getMaxSize        func() int
+	lah               lastAccessHeap
+}
+
+type cacheEntry struct {
+	lastAccessTime uint64
+	heapIdx        int
+
+	key   string
+	value interface{}
+}
+
+// NewCache returns new Cache with limited entries count
+func NewCache(getMaxSize func() int) *Cache {
+	c := &Cache{
+		m:                 make(map[string]*cacheEntry),
+		perKeyMisses:      make(map[string]int),
+		getMaxSize:        getMaxSize,
+		cleanerStoppedCh:  make(chan struct{}),
+		cleanerMustStopCh: make(chan struct{}),
+	}
+	go c.cleaner()
+	return c
+}
+
+// MustStop frees up resources occupied by c.
+func (c *Cache) MustStop() {
+	close(c.cleanerMustStopCh)
```

Comment:
should we wrap it with `sync.Once` to avoid panic if called twice?

Reply:
I think, it's a bug, if this method called twice